ad_block_client.cc

```diff
@@ -49,6 +49,52 @@ const int AdBlockClient::kFingerprintSize = 6;
 
 static HashFn2Byte hashFn2Byte;
 
+/**
+ * Checks to see if a URL is "blockable".
+ *
+ * Blockable URLs are ones that use one of the following protocols (any of
+ * which can be prefixed by "blob:")
+ *  - http
+ *  - https
+ *  - ws
+ *  - wss
+ */
+bool isBlockableProtocol(const char *url, int urlLen) {
```

Comment:
I think we should add a mini state machine for this to avoid the allocations.  Similar to how the normal parsing works with keeping an enum for the current state and the current char that is being processed.